From 812c7341f51e1c0a61b815a4fec1fe0834eabc01 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:37:56 -0400
Subject: [virt] kvm/nvmx: Replace kvm_set_cr0 with vmx_set_cr0 in load_vmcs12_host_state

Message-id: <1406040016-3289-82-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86160
O-Subject: [RHEL7 PATCH v2 081/221] KVM: nVMX: Replace kvm_set_cr0 with vmx_set_cr0 in load_vmcs12_host_state
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jan Kiszka <jan.kiszka@siemens.com>

kvm_set_cr0 performs checks on the state transition that may prevent
loading L1's cr0. For now we rely on the hardware to catch invalid
states loaded by L1 into its VMCS. Still, consistency checks on the host
state part of the VMCS on guest entry will have to be improved later on.

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
Reviewed-by: Gleb Natapov <gleb@redhat.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit 9e3e4dbf4469ba19971fbd547f1be713604c7f5e)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/vmx.c b/arch/x86/kvm/vmx.c
index 2a1a9fc..98a9a1a 100644
--- a/arch/x86/kvm/vmx.c
+++ b/arch/x86/kvm/vmx.c
@@ -8200,7 +8200,7 @@ static void load_vmcs12_host_state(struct kvm_vcpu *vcpu,
   * fpu_active (which may have changed).
   * Note that vmx_set_cr0 refers to efer set above.
   */
- kvm_set_cr0(vcpu, vmcs12->host_cr0);
+ vmx_set_cr0(vcpu, vmcs12->host_cr0);
  /*
   * If we did fpu_activate()/fpu_deactivate() during L2's run, we need
   * to apply the same changes to L1's vmcs. We just set cr0 correctly,
-- 
1.7.1